arangod/Aql/SubqueryEndExecutor.cpp

```diff
@@ -110,11 +111,20 @@ std::pair<ExecutionState, NoStats> SubqueryEndExecutor::produceRows(OutputAqlIte
           return {ExecutionState::WAITING, NoStats{}};
         }
         TRI_ASSERT(state == ExecutionState::DONE || state == ExecutionState::HASMORE);
-        TRI_ASSERT(shadowRow.isInitialized());
-        TRI_ASSERT(shadowRow.isRelevant());
+
+        if (state == ExecutionState::DONE && !shadowRow.isInitialized()) {
+          /* We had better not accumulated any results if we get here */
```

Comment:
It is not obvious to see what this case is about to handle and why it has to be like this.
Please add a comment to describe the state we reach here.